🎉 AIO Fighter - Full Auto Slayer BETA is Finally Here! 🎉

See1Duck opened the pull request:
## Slayer time!

I want to start by sincerely apologizing for the delays in delivering this long-awaited feature. Personal circumstances have slowed development more than anticipated, but I'm incredibly excited to finally release the **BETA version** of the Full Auto Slayer integration for the AIO Fighter plugin!

Since Slayer is notoriously slow to train, testing thoroughly by myself has been challenging. Now that it's open to many more users, expect bugs and quirks to surface. Please report any issues clearly to help me iron out problems swiftly.

Remember, this is just the **beginning**—lots of exciting new features are still planned for the future. Your patience, support, and feedback are deeply appreciated as we refine this tool together!

---

## 📜 Changelog - AIO Fighter Plugin (v2.0.0 BETA)

### 🚀 New Features

* **Full Auto Slayer Integration**

  * Slayer task management and navigation logic implemented.
  * Inventory setup automatically determined based on Slayer monsters.
  * Adaptive clustering algorithm added for accurate targeting.
  * Enhanced NPC interaction with antiban features and additional failsafes.
  * Improved banking logic and inventory management specifically for Slayer.

* **Overlay Enhancements**

  * Improved rendering of attackable areas and path visualization.
  * Pause and blacklist buttons added to the overlay.
  * Slayer-specific info displayed directly in overlay.

* **Auto Shopping (UNTESTED)**

  * Initial addition of auto-shopping functionality for Slayer supplies.

* **Prayer and Potion Management**

  * Enhanced prayer handling based on NPC combat levels.
  * MIXED\_LAZY\_FLICK option added for prayer styles.
  * Improved potion drinking logic and handling of empty vials.

* **New Monsters and Task Entries**

  * Crab variants included as new Slayer task entries.

### 🛠️ Improvements and Refactors

* Replaced general NPC usage with optimized `Rs2NpcModel`.
* Simplified monster tracking and prayer flicking logic.
* Updated configuration settings:

  * Slayer-specific configurations introduced.
  * Removed obsolete banking configuration options.
* Food consumption logic streamlined; deprecated methods marked for future updates.
* Improved attack style mapping accuracy and added logging for unknown styles.
* Enhanced logging throughout Slayer, NPC, and combat management.
* Cannon script and safety scripts enhanced with proper shutdown methods.
* Updated monster data and NPC attack style information.
* Removed old and redundant constants file.

### 🐞 Bug Fixes

* Looting logic now correctly checks player status.
* Xeric’s Honour teleportation temporarily disabled until further notice.

---

Thank you all for being part of this journey! 🚀 Let's conquer Slayer together!

See1Duck committed:
Rs2Bank:
refactor: clean up whitespace and formatting in bank handling methods

See1Duck committed:
GrandExchange:
feat: add method to count available Grand Exchange slots

See1Duck committed:
refactor(Rs2NpcManager): replace logging with Microbot logging utility and improve location logging
refactor(Rs2NpcManager): improved and updated the getClosestLocation

See1Duck committed:
aiofighter: feat(MonsterLocation): add method to get best cluster center for coordinates

See1Duck committed:
slayer:
refactor(Rs2Slayer): improve logging and handle empty slayer task scenario

See1Duck committed:
Task: feat(Task): add Crab entry with multiple variants

See1Duck committed:
Updated monster data & npc attack style data

See1Duck committed:
nmz:
refactor(PrayerPotionScript): move script to nmz package and simplify method signature

See1Duck committed:
microbot:
refactor(PathfinderConfig): simplify tab switching and restriction checks

See1Duck committed:
teleportation_items:
fix: comment out the entry for Xeric's talisman: Xeric's Honour until we can find a solution

See1Duck committed:
AIOFighterPlugin:
feat(Slayer): add slayer task management and improve error handling
chore(): updated version to 2.0.0 BETA

See1Duck committed:
AIOFighterPlugin:
feat(Overlay): enhance rendering of attackable areas and improve path visualization

See1Duck committed:
AIOFighterPlugin:
feat(Overlay): add pause and blacklist buttons to AIOFighterInfoOverlay
feat(Overlay): add Slayer info to AIOFighterInfoOverlay

See1Duck committed:
AIOFighterPlugin:
refactor(config): update config group name and adjust section visibility
feature(config): add config for slayer
chore(config): remove old banking config

See1Duck committed:
AIOFighterPlugin:
feat(shop): add ShopItem, ShopScript, and ShopType for auto shopping functionality for slayer (UNTESTED)

See1Duck committed:
AIOFighterPlugin:
feat(SafetyScript): add shutdown method to enhance plugin termination process

See1Duck committed:
AIOFighterPlugin:
refactor(Monster): replace NPC with Rs2NpcModel in Monster class

See1Duck committed:
AIOFighterPlugin:
feat(PointClusterer): add adaptive clustering algorithm for WorldPoint coordinates

See1Duck committed:
AIOFighterPlugin:
feat(InventorySetupUtil): add inventory setup determination for slayer monsters

See1Duck committed:
AIOFighterPlugin:
fix(LootScript): replace combat check with player status and correct comment typo

See1Duck committed:
AIOFighterPlugin:
fix(AttackStyleMapper): improve attack style mapping and add logging for unrecognized styles
feat(PrayerStyle): add MIXED_LAZY_FLICK option to prayer styles

See1Duck committed:
chore(aio fighter): removed old constants file

See1Duck committed:
AIOFighterPlugin:
feat(CannonScript): add shutdown method override

See1Duck committed:
AIOFighterPlugin:
refactor(FoodScript): remove empty vial check and streamline food consumption logic

TODO: replace deprecated methods

See1Duck committed:
AIOFighterPlugin:
feat(PotionManager): enhance potion drinking logic and manage empty vials

See1Duck committed:
AIOFighterPlugin:
feat(SlayerScript): add slayer task handling and navigation logic

See1Duck committed:
AIOFighterPlugin:
feat(PrayerScript): enhance prayer management based on NPC combat level

See1Duck committed:
AIOFighterPlugin:
refactor(FlickerScript): simplify monster tracking and prayer flicking logic

See1Duck committed:
AIOFighterPlugin:
feat(AttackNpcScript): enhance NPC interaction and add antiban features and some slayer failsafe measures

See1Duck committed:
AIOFighterPlugin:
feat(BankerScript): enhance banking logic and inventory management and support for slayer handling

See1Duck committed:
Merge branch 'development' into development

Gage307 commented:
Amazing! This must've been a lot of work; can't wait to use it!

See1Duck committed:
AIOFighterPlugin:
fix(BankerScript): improve protective equipment checks and add bypass logic for slayer helmet
fix(BankerScript): prevents endless idle loop if no transport items was needed

See1Duck committed:
Merge branch 'chsami:development' into development

Voxsylvae commented:
Amazing work, how you put all the things together, build in the slayer functionality into the AIO Fighter. Looking for forward to test it out.

See1Duck committed:
Merge branch 'development' into development

See1Duck committed:
AIOFighterPlugin: fix(transports): correct climb rock transport in Mount Karuulm Dungeon in transports.tsv

See1Duck committed:
AIOFighterPlugin: fix(npcs): update attack styles for NPCs 8610 and 8611

See1Duck committed:
`refactor(BankerScript): update handling of missing item quantities and improve inventory checks`
- Refactor transport item handling to use `Map` for item IDs and quantities.
- Add precise inventory quantity checks in `handleTeleports`.
- Improve inventory behavior by adding wait for changes after banking action.
- Add specific handling for "Rock hammer" and "Rock thrownhammer".

See1Duck committed:
`feat(State): add GETTING_TASK and INITIALIZING states`

See1Duck committed:
`feat(AIOFighterOverlay): display attack style on overlay`

See1Duck committed:
`feat(AIOFighterPlugin): add logging for inventory setup and support mixed lazy flick prayer style`

See1Duck committed:
feat(FlickerScript): add support for MIXED_LAZY_FLICK prayer style

See1Duck committed:
`feat(PotionManagerScript): add task state check to prevent script execution`

See1Duck committed:
feat(PrayerScript): improve prayer handling logic
- Remove redundant login and super.run() checks
- Optimize NPC selection by using findFirst() and null check
- Add checks to prevent overlapping prayer activations

See1Duck committed:
refactor(SlayerScript): remove unused variables and redundant methods
- Deleted `runOnce` and `test` variables
- Removed `walkToCurrentSlayerTask` method
- Replaced inline variable declaration with specific type for `monsterLocation`
- Added state update for obtaining new slayer tasks

See1Duck committed:
`fix(SlayerTaskMonster): add missing gear requirement for Wyrm`

See1Duck committed:
refactor(FlickerScript): replace list with AtomicReference for thread-safe updates
- Use AtomicReference for `currentMonstersAttackingUs` and `npcs` to ensure thread-safety
- Refactor methods to accommodate AtomicReference updates
- Simplify prayer flick logic and streamline NPC management

refactor(AIOFighterOverlay): update to use AtomicReference for monsters
- Replace `currentMonstersAttackingUs` with `currentMonstersAttackingUsRef` for consistency

See1Duck commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/aiofighter/combat/FlickerScript.java:
```diff
@@ -29,16 +29,16 @@
-    public static final AtomicReference<List<Monster>> currentMonstersAttackingUs = new AtomicReference<>(new ArrayList<>());
-    private final AtomicReference<List<Rs2NpcModel>> npcs = new AtomicReference<>(new ArrayList<>());
-
+    public static List<Monster> currentMonstersAttackingUs = new ArrayList<>();
```
This was just old code that didn't get merged properly. it has been addressed here [Commit cd141587](https://github.com/chsami/Microbot/commit/cd141587a3aefe572bd04d865bbbac6c1a2f255e)

See1Duck commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/aiofighter/combat/AttackNpcScript.java:
```diff
@@ -101,24 +115,65 @@ public void run(AIOFighterConfig config) {
+                                    .mapToAttackStyle(Rs2NpcManager.getAttackStyle(npc.getId()));
+                            if (attackStyle != null) {
+                                switch (attackStyle) {
+                                    case MAGE:
```
Its more targeted for the lazy flick to avoid taking a inital hit before the flicking is activated

See1Duck committed:
Merge branch 'chsami:development' into development
